packages/cli/src/commands/test.ts

```diff
@@ -233,10 +242,15 @@ export default class Test extends AuthCommand {
       reporters.forEach(r => r.onSchedulingDelayExceeded())
     })
 
-    runner.on(Events.CHECK_SUCCESSFUL, (checkRunId, check, result, links?: TestResultsShortLinks) => {
+    runner.on(Events.CHECK_SUCCESSFUL, async (checkRunId, check, result, links?: TestResultsShortLinks) => {
```

Comment:
The even emitter doesn't support async operations. This will lead to race conditions